app.rb

```diff
@@ -15,18 +17,31 @@
   require 'aws-sdk-resources'
   require_relative 's3'
   require_relative 'file_manager'
+  require_relative 'species'
 
   filename = "test_#{params[:id]}.csv"
   @rows = []
 
   FileManager.with_file(filename) do
     S3.new.get_file(filename)
     fullname = "#{S3::BASE_PATH}#{filename}"
+    @species = []
+    @cr = []
+    @en = []
+    @vu = []
     CSV.foreach(fullname, headers: true) do |row|
       @rows << row
+      @species << row
+      if row["redlist_status"] == "CR"
```

Comment:
The if statements are doing what they are supposed to do, but potentially you could make use of the ruby `group_by` function to group all species by `redlist_status` once they have all been collected into the `@species` variable.

For example:

```
@species.group_by { |hash| hash['redlist_status'] }
```

The above will return an hash with the redlist statuses as keys, like:

```
{
  'CR': [{species_name: 'name', common_name: 'name', redlist_status: 'CR', ...}, { ... }, ...],
  'EN': [{species_name: 'name', common_name: 'name', redlist_status: 'EN', ...}, { ... }, ...],
   ...
}
```

So basically, each key is a redlist status and the values are arrays with the CSV rows

Reply:
Thanks!